src/core/Crust/Plugins/api.ts

```diff
@@ -0,0 +1,511 @@
+            return (key: string) => {
+              const promise = clientStorage.getAsync(
+                (plugin?.extensionType === "widget"
+                  ? widget?.()?.id
+                  : plugin?.extensionType === "block"
+                  ? block?.()?.id
+                  : "") ?? "",
+                key,
+              );
+              promise.finally(() => {
+                startEventLoop?.();
+              });
+              return promise;
+            };
+          },
+          get setAsync() {
+            return (key: string, value: any) => {
+              const localValue =
+                typeof value === "object" ? JSON.parse(JSON.stringify(value)) : value;
+              const promise = clientStorage.setAsync(
+                (plugin?.extensionType === "widget"
+                  ? widget?.()?.id
+                  : plugin?.extensionType === "block"
+                  ? block?.()?.id
+                  : "") ?? "",
+                key,
+                localValue,
+              );
+              promise.finally(() => {
+                startEventLoop?.();
+              });
+              return promise;
+            };
+          },
+          get deleteAsync() {
+            return (key: string) => {
+              const promise = clientStorage.deleteAsync(
+                (plugin?.extensionType === "widget"
+                  ? widget?.()?.id
+                  : plugin?.extensionType === "block"
+                  ? block?.()?.id
+                  : "") ?? "",
+                key,
+              );
+              promise.finally(() => {
+                startEventLoop?.();
+              });
+              return promise;
+            };
+          },
+          get keysAsync() {
+            return () => {
+              const promise = clientStorage.keysAsync(
+                (plugin?.extensionType === "widget"
+                  ? widget?.()?.id
+                  : plugin?.extensionType === "block"
+                  ? block?.()?.id
+                  : "") ?? "",
+              );
+              promise.finally(() => {
+                startEventLoop?.();
+              });
+              return promise;
+            };
+          },
+        },
+        ...events,
+      },
+      plugin?.extensionType === "block"
+        ? {
+            get layer() {
+              return layer?.();
+            },
+            get block() {
+              return block?.();
+            },
+          }
+        : {},
+      plugin?.extensionType === "widget"
+        ? {
+            get widget() {
+              return {
+                ...widget?.(),
+                moveTo: (options: WidgetLocationOptions) => {
+                  const widgetId = widget?.()?.id;
+                  if (!widgetId) return;
+                  moveWidget?.(widgetId, options);
+                },
+              };
+            },
+          }
+        : {},
+    ),
+  });
+}
+
+export function commonReearth({
```

Comment:
Basically I copied from current layer system, but this function is changed to combine new layer props.